File: Python/Car.py
from Vehicle import Vehicle


class Car(Vehicle):
    def __init__(self, license_plate, brand, year, color, num_seats, num_doors):
        super().__init__(license_plate, brand, year, color)
        self.__num_seats = num_seats
        self.__num_doors = num_doors

    def get_num_seats(self):
        return self.__num_seats

    def set_num_seats(self, num_seats):
        self.__num_seats = num_seats

    def get_num_doors(self):
        return self.__num_doors

    def set_num_doors(self, num_doors):
        self.__num_doors = num_doors

File: CPP/Car.cpp
/*
Bismillah
Saya Themy Sabri Syuhada dengan NIM 2203903.
Demi keberkahan-Nya, saya berjanji mengerjakan
latihan praktikum 4 DPBO dengan jujur
dan tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/

#pragma once

#include <iostream>
#include <string>
#include "Vehicle.cpp"

using namespace std;

class Car : public Vehicle
{
private:
    int num_seats;
    int num_doors;

public:
    // Constructor
    Car(std::string license_plate, std::string brand, int year, std::string color, int num_seats, int num_doors)
        : Vehicle(license_plate, brand, year, color), num_seats(num_seats), num_doors(num_doors) {}

    // Getter and Setter for number of seats
    int get_num_seats()
    {
        return num_seats;
    }

    void set_num_seats(int new_num_seats)
    {
        num_seats = new_num_seats;
    }

    // Getter and Setter for number of doors
    int get_num_doors()
    {
        return num_doors;
    }

    void set_num_doors(int new_num_doors)
    {
        num_doors = new_num_doors;
    }
};